pilot/cmd/pilot-agent/main.go

```diff
@@ -476,13 +479,19 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
-// check if SDS UDS path and token path exist, if both exist, requests key/cert
-// using SDS instead of secret mount.
+// detectSds checks config for control plane SDS . The returned values are used by caller
+// to decide whether SDS or file-mounted secret is used for control plane.
```

Comment:
nit: to decide whether SDS is used for the connection to Pilot.

Reply:
It is not only about connection to Pilot, but also communication between control plane component sidecars. In https://github.com/istio/istio/pull/16459, all the template changes under pilot/docker/ are relevant to returned values from detectSds().